### ISSUE TITLE
Consider moving from submodules to subtree

### ISSUE THREAD

**Issue by sharwell** [gitextensions/gitextensions#4616]
Currently there are three external repositories included as submodules:

1. Git.hub
1. NBug
1. conemu-inside

While submodules are great for submitting changes back upstream (to the original repositories), they introduce complexity when changes need to be made. If there is no intent to submit the changes back upstream, I recommend modifying the repository to use subtrees instead of submodules. This change would have the following overall impact on workflow:

* Changes could be made to the embedded externals without worrying about submitting those changes to multiple repositories
* Changes can still be pulled from upstream when necessary
* It would be difficult to submit our changes back to the upstream repository

In other words, this change favors local iteration over the ability to send changes back to the original repository. If this is desirable, I can send a pull request to implement the change.

**Comment by sharwell** [comment]
💡 I implemented the change for one of the three externals to review the impact.

**Comment by RussKie** [comment]
We do occasional work on submodules. For example, every time we bump the version GEDocs is updated as well. Git.hub has received some updates lately as well.

If anything I would decouple project dependencies and consume them via assemblies (nuget packages?)

**Comment by sharwell** [comment]
@RussKie This change would not impact the ability to consume changes made in the original repositories. It would only impact the ability to submit those changes back.

Also note that GitExtensionsDocs is less relevant here. I'm mostly interested in the other three.

**Comment by gerhardol** [comment]
I rather keep submodules, clearly separating the external dependencies.
It is quite minor job we do on the submodules.

Submodules is what GE is best at, really why I use GE.
Subtrees would maybe make changes easier for those not experienced with submodules, but there are no direct support for subtrees in GE and less general knowledge how to handle subtrees.

**Comment by RussKie** [comment]
> It would only impact the ability to submit those changes back.

And that's what I am worried about. To me subtree feels like more maintenance, which is something I like to minimise.

**Comment by sharwell** [comment]
Subtrees make it easier to deal with cross-cutting concerns, such as updating ConEmuWinForms as part of #4601, or switching to `PackageReference`-style NuGet dependencies to simplify the code base.

**Comment by vbjay** [comment]
They also put you in the hotseat for making sure your commits don't mix. We
have and do submit changes upstream.

**Comment by RussKie** [comment]
Thank you, Sam. But at this stage it is not something we wish to pursue further.
